mirrord/intproxy/src/proxies/simple.rs

```diff
@@ -217,8 +217,34 @@ impl BackgroundTask for SimpleProxy {
                             .await;
                     }
                 }
-                SimpleProxyMessage::FileReq(message_id, session_id, req) => {
-                    self.file_reqs.insert(message_id, session_id);
+                // TODO(alex): We can remove this case when users are up-to-date for `readlink`.
+                SimpleProxyMessage::FileReq(
+                    message_id,
+                    layer_id,
+                    req @ FileRequest::ReadLink(_),
+                ) => {
+                    if protocol_version
+                        .as_ref()
+                        .is_some_and(|version| READDIR_BATCH_VERSION.matches(version))
+                    {
+                        self.file_reqs.insert(message_id, layer_id);
```

Comment:
This means that if user calls readlink, they'll just crash - that's not what we want

Reply:
My bad. I changed it to `Bypass` on `NotImplemented` error. This way we avoid having to check the version in the layer.